src/System.IO.FileSystem/tests/FileStream/Dispose.cs

```diff
@@ -58,6 +63,92 @@ protected override void Dispose(bool disposing)
             }
         }
 
+
+        [Fact]
+        public void Dispose_ThrowDuringFlushWriteBuffer_TrueArgThrows()
+        {
+            RemoteInvoke(() =>
+            {
+                string fileName = GetTestFilePath();
+                using (FileStream fscreate = new FileStream(fileName, FileMode.Create))
+                {
+                    fscreate.WriteByte(0);
+                }
+                bool writeDisposeInvoked = false;
+                Action<bool> writeDisposeMethod = (disposing) =>
+                {
+                    writeDisposeInvoked = true;
+                };
+                using (var fsread = new FileStream(fileName, FileMode.Open, FileAccess.Read))
+                {
+                    Action act = () => // separate method to avoid JIT lifetime-extension issues
+                    {
+                        using (var fswrite = new MyFileStream(fsread.SafeFileHandle, FileAccess.Write, writeDisposeMethod))
+                        {
+                            fswrite.WriteByte(0);
+
+                            // Normal dispose should call Dispose(true). Throws due to FS trying to flush write buffer
+                            Assert.Throws<UnauthorizedAccessException>(() => fswrite.Dispose());
+                            Assert.True(writeDisposeInvoked, "Expected Dispose(true) to be called from Dispose()");
+                            writeDisposeInvoked = false;
+
+                            // Only throws on first Dispose call
+                            fswrite.Dispose();
+                            Assert.True(writeDisposeInvoked, "Expected Dispose(true) to be called from Dispose()");
+                            writeDisposeInvoked = false;
+                        }
+                        Assert.True(writeDisposeInvoked, "Expected Dispose(true) to be called from Dispose() again");
+                        writeDisposeInvoked = false;
+                    };
+                    act();
+
+                    for (int i = 0; i < 2; i++)
+                    {
+                        GC.Collect();
+                        GC.WaitForPendingFinalizers();
+                    }
+                    Assert.False(writeDisposeInvoked, "Expected finalizer to have been suppressed");
+                }
+            }).Dispose();
+        }
+
+        [Fact]
+        public void Dispose_ThrowDuringFlushWriteBuffer_FalseArgWontThrow()
+        {
+            RemoteInvoke(() =>
```

Comment:
Same question

Reply:
Before the fix is merged the child process here throws an exception on finalization.
In order to make sure the test fails before coreclr PR is merged, I will need to return SuccessExitCode" so to make sure the remote task executor asserts no exceptions were thrown in the child process.

The reason I added RemoteInvoke was to make sure if the exception is thrown it doesnt kill the main xunit process.

The actual unit test that exercises the bugfix is ```Dispose_ThrowDuringFlushWriteBuffer_FalseArgWontThrow``` here.
the other unit test, ```Dispose_ThrowDuringFlushWriteBuffer_TrueArgThrows```, already returns SuccessExitCode even before the bugfix.